# London Airbnb Dataset EDA
___

### Importación de librerías

In [2]:
# Librerías de análisis de datos
import numpy as np
import pandas as pd

# Librerías de visualización
import matplotlib.pyplot as plt
import seaborn as sns

### Exploración básica del dataset

Estas son las variables por entrada que contiene el dataset:
+ **id**: número de identificación de la publicación del alojamiento.
+ **name**: nombre de la publicación del alojamiento. 
+ **host_id**: número de identificación del anfitrión.
+ **host_name**: nombre del anfitrión.
+ **neighbourhood_group**: grupo de barrios.
+ **neighbourhood**:barrio o vecindario. 
+ **latitude**: latitud de la ubicación del alojamiento.
+ **longitude**:longitud de la ubicación del alojamiento.
+ **room_type**: tipo de habitación.
+ **price**: precio.
+ **minimum_nights**: noches mínimas.
+ **number_of_reviews**: cuantas personas han revisado el anuncio.
+ **last_review**: última revisión de la publicación.
+ **reviews_per_month**: promedio de revisión por mes.
+ **calculated_host_listings_count**: recuento de host calculados por un propietario/a.
+ **availability_365**: disponibilidad en los 365 días. 

In [7]:
# Una variable para la ruta, buenas prácticas
path_to_data = "./london_airbnb.csv"

# Importamos el dataset y comprobamos que está correcto
df = pd.read_csv(path_to_data)
df.head()

id                                               name  host_id  \
0  11551         Arty and Bright London Apartment in Zone 2    43039   
1  13913                Holiday London DB Room Let-on going    54730   
2  90700                  Sunny Notting Hill flat & terrace   491286   
3  15400                Bright Chelsea  Apartment. Chelsea!    60302   
4  92399  MODERN SELF CONTAINED ARCHITECT FLATLET, ISLIN...   497366   

       host_name  neighbourhood_group           neighbourhood  latitude  \
0        Adriano                  NaN                 Lambeth  51.46225   
1          Alina                  NaN               Islington  51.56802   
2           Chil                  NaN  Kensington and Chelsea  51.51074   
3       Philippa                  NaN  Kensington and Chelsea  51.48796   
4  Andrea & Mark                  NaN               Islington  51.55071   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0   -0.11732  Entire home/apt     88               3                185   
1   -0.11121     Private room     65               1                 19   
2   -0.19853  Entire home/apt    105               2                339   
3   -0.16898  Entire home/apt    100              30                 88   
4   -0.08547     Private room     77               1                207   

  last_review  reviews_per_month  calculated_host_listings_count  \
0  2019-09-15               1.58                               2   
1  2019-10-07               0.17                               2   
2  2019-07-30               3.33                               2   
3  2019-09-23               0.73                               1   
4  2019-10-21               2.04                               2   

   availability_365  
0               336  
1               365  
2               268  
3               158  
4               336

In [9]:
# Se muestran las últimas entradas.
df.tail()

id                                               name    host_id  \
85063  38398365                   Beautiful flat in Central London   24782379   
85064  39869123    Elegant stylish modern house with amazing view!   21145791   
85065  39869214                  Holland Road - Kensington Olympia  251922260   
85066  39869249     Functional single room 18 min to London Bridge  255372486   
85067  39869282  EXQUISITE APARTMENT, PEACEFUL ENVIRONMENT IN SOHO  210531062   

       host_name  neighbourhood_group           neighbourhood  latitude  \
85063    Shabnam                  NaN                 Lambeth  51.48601   
85064     Ingrid                  NaN                  Merton  51.40628   
85065  Christian                  NaN  Hammersmith and Fulham  51.50244   
85066     Nudrat                  NaN                 Bromley  51.35742   
85067       Radi                  NaN             Westminster  51.51090   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
85063   -0.12766  Entire home/apt    117               3                  2   
85064   -0.23244     Private room     69               2                  0   
85065   -0.21547  Entire home/apt     90               3                  0   
85066    0.10764     Private room     30               2                  0   
85067   -0.12480  Entire home/apt    187               1                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
85063  2019-11-04                2.0                               1   
85064         NaN                NaN                               2   
85065         NaN                NaN                               4   
85066         NaN                NaN                               4   
85067         NaN                NaN                               1   

       availability_365  
85063                59  
85064               113  
85065                63  
85066                 1  
85067               311

In [10]:
#Cantidad de entradas y columnas. 
df.shape

(85068, 16)

Se observa: 
+ 16 columnas. 
+ 85068 entradas.

In [8]:
# Información sobre el dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 85068 entries, 0 to 85067
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              85068 non-null  int64  
 1   name                            85043 non-null  object 
 2   host_id                         85068 non-null  int64  
 3   host_name                       85056 non-null  object 
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   85068 non-null  object 
 6   latitude                        85068 non-null  float64
 7   longitude                       85068 non-null  float64
 8   room_type                       85068 non-null  object 
 9   price                           85068 non-null  int64  
 10  minimum_nights                  85068 non-null  int64  
 11  number_of_reviews               85068 non-null  int64  
 12  last_review                     

Se observa:
+ La columna "neighbourhood_group" se encuentra vacía por lo que no se considerará por falta de datos.
+ EL resto de columnas, en su mayoría, se encuentran completas. A excepción de las columnas: reviews_per_month, last_review, name y host_name.

In [12]:
#Conteo de datos faltantes por columna. 
df.isnull().sum(axis = 0)

id                                    0
name                                 25
host_id                               0
host_name                            12
neighbourhood_group               85068
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       20006
reviews_per_month                 20006
calculated_host_listings_count        0
availability_365                      0
dtype: int64

Se observa:
+ Las columnas con más datos faltantes son last_review y reviews_per_month, la cantidad es significativa. 

### Creación de nueva tabla
___

In [14]:
# Creando nueva tabla "df_londres" descartando columnas:
df_londres = df.drop(columns=["neighbourhood_group","id", "name","host_id","host_name"])
df_londres.head()

neighbourhood  latitude  longitude        room_type  price  \
0                 Lambeth  51.46225   -0.11732  Entire home/apt     88   
1               Islington  51.56802   -0.11121     Private room     65   
2  Kensington and Chelsea  51.51074   -0.19853  Entire home/apt    105   
3  Kensington and Chelsea  51.48796   -0.16898  Entire home/apt    100   
4               Islington  51.55071   -0.08547     Private room     77   

   minimum_nights  number_of_reviews last_review  reviews_per_month  \
0               3                185  2019-09-15               1.58   
1               1                 19  2019-10-07               0.17   
2               2                339  2019-07-30               3.33   
3              30                 88  2019-09-23               0.73   
4               1                207  2019-10-21               2.04   

   calculated_host_listings_count  availability_365  
0                               2               336  
1                               2               365  
2                               2               268  
3                               1               158  
4                               2               336

In [23]:
# Se comprueba la eliminación de columnas.
df_londres.shape

(85068, 11)

Se observa:
+ Se redujo de 16 a 11 columnas.

In [15]:
# Información nueva tabla "df_londres":
df_londres.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 85068 entries, 0 to 85067
Data columns (total 11 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   neighbourhood                   85068 non-null  object 
 1   latitude                        85068 non-null  float64
 2   longitude                       85068 non-null  float64
 3   room_type                       85068 non-null  object 
 4   price                           85068 non-null  int64  
 5   minimum_nights                  85068 non-null  int64  
 6   number_of_reviews               85068 non-null  int64  
 7   last_review                     65062 non-null  object 
 8   reviews_per_month               65062 non-null  float64
 9   calculated_host_listings_count  85068 non-null  int64  
 10  availability_365                85068 non-null  int64  
dtypes: float64(3), int64(5), object(3)
memory usage: 7.1+ MB


In [16]:
# Datos faltantes en porcentaje ordenado de mayor a menor.
df_londres.isnull().mean().sort_values(ascending = False)

last_review                       0.235177
reviews_per_month                 0.235177
neighbourhood                     0.000000
latitude                          0.000000
longitude                         0.000000
room_type                         0.000000
price                             0.000000
minimum_nights                    0.000000
number_of_reviews                 0.000000
calculated_host_listings_count    0.000000
availability_365                  0.000000
dtype: float64

Se observa:
+ Falta de datos en las columnas last_review y reviews_per_month. Mismo porcentaje= 23.5177%.

In [ ]:
# Identificación de variables:
categoricas = ["neighbourhood", "room_type", "last_review"]
numericas = ["latitude", "longitude", "price", "minimum_nights", "number_of_reviews", "reviews_per_month", "calculated_host_listings_count", "availability_365"]